src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -537,9 +537,18 @@ private async Task<bool> ContinueActivatingLicense(License license, bool skipLea
                     // the features of the license, even before we check with the
                     // server
                     SetLicense(license.Id, newLicenseStatus.Attributes);
-
-                    // license expired, we'll try to update it
-                    license = await GetUpdatedLicenseInternal(license);
+                    try
+                    {
+                        // license expired, we'll try to update it
+                        license = await GetUpdatedLicenseInternal(license);
+                    }
+                    catch (Exception e)
+                    {
+                        if (e is HttpRequestException)
```

Comment:
Include IOException / SocketException here as well?

Reply:
Probably not needed? https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=netcore-3.1#System_Net_Http_HttpClient_SendAsync_System_Net_Http_HttpRequestMessage_ but maybe we should include `TaskCanceledException` ?

Reply:
Absolutely `TaskCanceledException`, but...

```
var client = new HttpClient();
client.GetAsync("http://e.example.com").GetAwaiter().GetResult();
```

Gives:

```
Unhandled exception. System.Net.Http.HttpRequestException: No such host is known.
 ---> System.Net.Sockets.SocketException (11001): No such host is known.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
```

Reply:
Yes, but the 'SocketException' is an inner ex of HttpRequestException?

Reply:
Hm... that is an inner exception, never mind.

Reply:
Do we want such deep error handling ? It still will throw a LicenseExpiredException with inner exception if it is not a HttpRequestException

Reply:
@garayx what you did is fine, I think